lib/puppet/provider/vc_vm_group/default.rb

```diff
@@ -73,34 +73,29 @@ def vm_group
   def vm_list(vm_names)
     @vms ||= 
       begin  
-        vms = find_rp_vms(cluster.resourcePool, vm_names)
+        vms = find_vms(cluster, vm_names)
         vms.uniq { |vm| vm._ref}
       end
   end
 
   def match_vm(matches, vm)
     matched_vms = []
     matches.each do |match|
-      matched_vms << vm if vm.name =~ /#{match}/
+      matched_vms << vm if vm.name == match
     end
     matched_vms
   end
 
- def find_rp_vms(resourcepool,vm_names)
-    @vm_obj ||= []
-    resourcepool.childConfiguration.each do |child|
-      case child.entity
-      when RbVmomi::VIM::VirtualMachine
-        @vm_obj << match_vm(vm_names, child.entity) 
-      when RbVmomi::VIM::VirtualApp
-        find_rp_vms(child.entity, vm_names)
-      when RbVmomi::VIM::ResourcePool
-        find_rp_vms(child.entity, vm_names)
-      else
-        Puppet.warning "#{self} find_rp_vm: unknown child type found: #{child.entity.class}"
-      end
+  def find_vms(container, vm_names)
+    vm_obj = []
+    vmview = vim.serviceContent.viewManager.CreateContainerView(
+      recursive: true, container: container,
+      type: ['VirtualMachine'])
+    vmview.view.each do |vm|
+      vm_obj << match_vm(vm_names, vm)
     end
-    @vm_obj.flatten
+    vmview.DestroyView
+    vm_obj.flatten
```

Comment:
vm_obj is an array of VM objects, not just names. Using intersection would just return an array of names.
I'll have to think if there are better ways to find the VM objects.